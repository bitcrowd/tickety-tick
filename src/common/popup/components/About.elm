module About exposing (..)

import Html exposing (..)
import Html.Attributes exposing (..)
import Html.Events exposing (..)

import ExternalLink

import Messages exposing (Msg(Navigate))
import Routes exposing (Route(IndexRoute))


view : Html Msg
view =
  div []
    [ div [ class "navbar navbar-light navbar-fixed-top navbar-tt" ]
      [ div [ class "nav navbar-nav" ]
        [ a [ class "nav-link", href "#index", onClick (Navigate IndexRoute) ] [ text "< back" ]
        ]
      ]
    , div [ class "content" ]
      [ div [ class "container" ]
        [ h1 [ class "h3 m-t-1 m-b-1" ]
          [ img [ src "../icons/icon-32.png", class "logo-sm", attribute "role" "presentation" ] []
          , text " tickety-tick"
          ]
        , h6 [] [ text "Usage:" ]
        , ol [ class "m-t-1 p-l-2" ]
            [ li [] [ text "Open one ore more tickets in your favourite ticket tracking system, then click the extension icon." ]
            , li [] [ text "You can then select if you want to create a commit message or a branch name based on these tickets." ]
            ]
        , p []
            [ text "This extension is open-source software by the fellows at "
            , extlink "http://bitcrowd.net" "bitcrowd"
            , text "."
            ]
        , p []
            [ text "The source code is available on "
            , extlink "https://github.com/bitcrowd/tickety-tick" "GitHub"
            , text "."
            ]
        , p [ class "small" ]
            [ text "Logo by "
            , extlink "http://thenounproject.com/term/ticket/92194/" "RamÃ³n G."
            , text " under CC-BY 3.0"
            , br [] []
            , text "Other icons from "
            , extlink "https://octicons.github.com/" "GitHub Octicons"
            ]
          ]
        ]
      ]

extlink = ExternalLink.view
