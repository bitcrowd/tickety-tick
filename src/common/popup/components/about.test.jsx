import React from 'react';
import { shallow } from 'enzyme';

import About from './about';

describe('about', () => {
  function render(/* overrides */) {
    return shallow(<About />);
  }

  it('renders', () => {
    expect(render({})).toMatchSnapshot();
  });
});
