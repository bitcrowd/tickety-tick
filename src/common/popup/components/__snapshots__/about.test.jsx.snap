// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`about renders 1`] = `
<div>
  <div
    className="navbar navbar-light fixed-top bg-white border-bottom"
  >
    <ul
      className="nav navbar-nav"
    >
      <li
        className="nav-item"
      >
        <Link
          className="nav-link"
          to="/"
        >
          &lt; back
        </Link>
      </li>
    </ul>
  </div>
  <div
    className="content"
  >
    <div
      className="container"
    >
      <h1
        className="h3 mt-3 mb-3"
      >
        <img
          alt=""
          className="logo-sm"
          src="icon-32.png"
        />
         
        tickety-tick
      </h1>
      <h6>
        Usage:
      </h6>
      <ol
        className="mt-2 pl-3"
      >
        <li>
          Open one or more tickets in your favourite ticket tracking system, then click the extension icon.
        </li>
        <li>
          You can then select if you want to create a commit message or a branch name based on these tickets.
        </li>
      </ol>
      <p>
        This extension is open-source software by the fellows at
         
        <ExternalLink
          href="http://bitcrowd.net"
        >
          bitcrowd
        </ExternalLink>
        .
      </p>
      <p>
        The source code is available on
         
        <ExternalLink
          href="https://github.com/bitcrowd/tickety-tick"
        >
          GitHub
        </ExternalLink>
        .
      </p>
      <p
        className="small"
      >
        <ExternalLink
          href="https://github.com/bitcrowd/tickety-tick/tree/test-commit-hash"
          title="Browse source code for this version"
        >
          test-commit-hash
        </ExternalLink>
      </p>
      <p
        className="small"
      >
        <span>
          Logo by 
        </span>
        <ExternalLink
          href="http://thenounproject.com/term/ticket/92194/"
        >
          Ramón G.
        </ExternalLink>
        <span>
           under CC-BY 3.0
        </span>
        <br />
        <span>
          Other icons from 
        </span>
        <ExternalLink
          href="https://octicons.github.com/"
        >
          GitHub Octicons
        </ExternalLink>
      </p>
    </div>
  </div>
</div>
`;
